buildSrc/src/main/kotlin/Compilations.kt

```diff
@@ -2,16 +2,11 @@
  * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
  */
 
-import org.jetbrains.kotlin.gradle.dsl.*
-import org.jetbrains.kotlin.gradle.plugin.*
+import org.jetbrains.kotlin.gradle.tasks.*
 
-fun KotlinCompilation<KotlinCommonOptions>.configureCompilation() {
-    kotlinOptions {
-//        if (platformType == KotlinPlatformType.jvm && !IDEA_ACTIVE) {
-//            allWarningsAsErrors = true
-//        }
-
-        freeCompilerArgs += "-opt-in=kotlin.RequiresOptIn"
-        freeCompilerArgs += "-Xexpect-actual-classes"
+fun KotlinCompilationTask<*>.configureCompilerOptions() {
+    compilerOptions {
+        progressiveMode.set(true)
+        freeCompilerArgs.addAll("-opt-in=kotlin.RequiresOptIn", "-Xexpect-actual-classes")
```

Comment:
```suggestion
        freeCompilerArgs.addAll("-Xexpect-actual-classes")
```

https://kotlinlang.org/docs/whatsnew17.html#stable-opt-in-requirements